addons/html_editor/controllers/main.py

```diff
@@ -547,3 +562,47 @@ def bus_broadcast(self, model_name, field_name, res_id, bus_data):
         bus_data.update({'model_name': model_name, 'field_name': field_name, 'res_id': res_id})
         request.env['bus.bus']._sendone(channel, 'editor_collaboration', bus_data)
 
+    @http.route('/html_editor/link_preview_external', type="json", auth="public", methods=['POST'])
+    def link_preview_metadata(self, preview_url):
+        return link_preview.get_link_preview_from_url(preview_url)
+
+    @http.route('/html_editor/link_preview_internal', type="json", auth="public", methods=['POST'])
```

Comment:
Do we need it to be public ?

Reply:
especially if we fetch the action in sudo

Reply:
Yes, I agree it'll be better to be `user`. I made it public cause I thought since the person already has the link, they should have the right to preview it even though not authenticated. But I didn't think about the sudo action fetch :pray: